### PR TITLE
Implementation

### PR DESCRIPTION
**Pull request by ethanwangtang**
The initial goal here was to setup my model and training functions since my data processing and Dataset class were already complete.

Ultimately, I updated my Dataset class to match existing training functions and wrote higher level code to train it. 

Unfortunately, my model written by chatGPT wasn't able to support the data so must be reworked. I tested a few things but ultimately need to dig further into pytorch model documentation without chatGPT to progress.

### CODE REVIEW

**Comment by ethanwangtang** [comment]
Prompts that generated Dataset before adjustments:

# -- prompt --
# me: Ok fuck the LSTM we are gonna use a transformer model. It makes perfect sense right? we have a limited number of tokens since the x and y coords are on a grid, there's only 4 types of notes which means token wise... like yeah. The only issue is the time between notes, since there's only like 400 and it's in 10ms strides, you're just not likely to see the same numbers that many times.
# me: Let's go step by step. The dataset, the x and y coords are already on that grid system, that's why in the example data you see 2, -3, -1, -1, 1, -1. We don't really need to do anything to it, just tokenize from ints.
# me: The time is also already in 10ms strides. And of course, the note_types is just 1 of 4 ints (0, 1, 2, 3)

**Comment by ethanwangtang** [comment]
Prompt that generated model:

prompt: Write for me a torch LSTM model that takes an input of xPos, yPos, noteType (which is one of 4 non-continuous ints), timeSinceLast in 10ms strides, and timeUntilNext in 10ms strides. For context, this is for a rhythm game map generator. The labels are the next xPos, yPos, and noteType in the sequence. Use an LSTM with 2 layers and 128 cells each.
